### PR TITLE
P service enclave

### PR DESCRIPTION
**Pull request by aaadesok**
Added initial PService Enclave functionality. Still need RSA and IAS verification.

### CODE REVIEW

File: pservice/pdo/pservice/pdo_enclave.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def create_signup_info(nonce):
+    # Part of what is returned with the signup data is an enclave quote, we
+    # want to update the revocation list first.
+    update_sig_rl()
+
+    # Now, let the enclave create the signup data
+    # signup_data = enclave.create_enclave_data(originator_public_key_hash)
```
Comment (byron-marohn) [inline pservice/pdo/pservice/pdo_enclave.py]: commented code

File: pservice/pdo/pservice/pdo_enclave.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def generate_enclave_secret(enclave_sealed_data, sealed_secret, enclave_id, contract_id, opk, enclave_key):
+ return enclave.generate_enclave_secret(enclave_sealed_data, sealed_secret, enclave_id, contract_id, opk, enclave_key)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def create_signup_info(nonce):
```
Comment (byron-marohn) [inline pservice/pdo/pservice/pdo_enclave.py]: Do we actually need all of this signup data stuff? The pservice enclave isn't actually registered with the ledger, unlike the eservice...

File: pservice/pdo/pservice/pdo_helper.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        """read_from_file -- read enclave data from a file and initialize a new
+        Enclave object with the resulting data.
+
+        :param file_name:  string, name of the file
+        :param search_path: list of strings, directories to search for the data file
+        :param txn_keys: object of type TransactionKeys
+        """
+
+        # if txn_keys is None :
+        #     txn_keys = keys.TransactionKeys()
```
Comment (byron-marohn) [inline pservice/pdo/pservice/pdo_helper.py]: commented code

File: pservice/pdo/pservice/pdo_helper.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    # -------------------------------------------------------
+    @classmethod
+    def create_new_enclave(cls) :
+        """create_new_enclave -- create a new enclave
+
+        :param txn_keys: object of type TransactionKeys
+        """
+
+        # if txn_keys is None :
+        #     txn_keys = keys.TransactionKeys()
```
Comment (byron-marohn) [inline pservice/pdo/pservice/pdo_helper.py]: commented code

File: pservice/pdo/pservice/pdo_helper.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+            enclave_info['proof_data'] = enclave_data.proof_data
+
+        return cls(enclave_info)
+
+    # -------------------------------------------------------
+    def __init__(self, enclave_info) :
+
+        # initialize the keys that can be used later to
+        # register the enclave with the ledger
+        # self.txn_keys = txn_keys
```
Comment (byron-marohn) [inline pservice/pdo/pservice/pdo_helper.py]: commented code

File: pservice/pdo/pservice/pdo_enclave.py (hunk 2 above)
Comment (aaadesok) [reply pservice/pdo/pservice/pdo_enclave.py]: It is not being used to register with enclave, just to create the enclave data. I can change the method names.

**Pull request merged**